hashstructure.go

```diff
@@ -2,12 +2,17 @@ package hashstructure
 
 import (
 	"encoding/binary"
+	"errors"
 	"fmt"
 	"hash"
 	"hash/fnv"
 	"reflect"
 )
 
+var (
+	ErrNotStringer = errors.New("field tag set to string, but struct does not implement fmt.Stringer")
```

Comment:
I think we should include the field name in the error output otherwise this can be pretty vague.